src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
```

Comment:
Make sure that the `List` interface is defined or imported correctly in your project. This is crucial for implementing the `ArrayList` class.